recyclercore-compiler/src/main/java/com/carrotcreative/recyclercore/compiler/ClassUtil.java

```diff
@@ -0,0 +1,22 @@
+package com.carrotcreative.recyclercore.compiler;
+
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+
+/**
+ * Created by kaushik on 9/30/16.
+ */
+
+class ClassUtil
+{
+    static String getClassName(TypeElement type, String packageName)
+    {
+        int packageLen = packageName.length() + 1;
+        return type.getQualifiedName().toString().substring(packageLen).replace('.', '$');
```

Comment:
Good catch, I believe it was for Inner classes. For example 
```
package app.ui
MainActivity {
  @RCModel
  TestModel {
  }
}
```
In this case the package would be `app.ui` and the class will be `MainActivity.TestModel` which this util function would change it into `MainActivity$TestActivity`

I just tried it and hit a roadblock, will need to test this and get back to you.